pkg/metrics/metrics.go

```diff
@@ -32,7 +32,13 @@ func StartMetrics(config metricsConfig) {
 	http.Handle(config.metricsPath, promhttp.Handler())
 	log.Info(fmt.Sprintf("Port: %s", config.metricsPort))
 	metricsPort := ":" + (config.metricsPort)
-	go http.ListenAndServe(metricsPort, nil)
+	go func() {
+		err := http.ListenAndServe(metricsPort, nil)
+		if err != nil {
+			// port not available error
+			log.Error(err, "Port not available")
```

Comment:
I think it would be better to say "Binding to port failed" since the error might be something else than it not being available. Since the error is being logged here anyway, a more generic error message make sense.